### ISSUE TITLE
SBE may freeze in some scenarios

### ISSUE THREAD

**Issue by ErikMogensen** [paolosalvatori/ServiceBusExplorer#196]
There is a bug in some Service Bus NuGet packages. According to https://www.nuget.org/packages/WindowsAzure.ServiceBus/4.1.8 the issue is:

> General: Fix a bug where application can hang when using WebSockets with a SynchronizationContext (WinForms, WPF)

Therefore SBE should be rebuilt with NuGet WindowsAzure.ServiceBus v4.1.8.

**Comment by SeanFeldman** [comment]
I'll take care of that.
Release notes are extremely vague. Wish the team would add details what clients are affected 🙂

**Comment by ErikMogensen** [comment]
I found that if I connect to a namespace using Connectivity Mode = Http or Https then issue appears. The issue also appeared if SBE was configured to use a proxy and Connectivity Mode = AutoDetect. So I believe every SBE which connects to SB through a proxy is affected, since it has to run in Http or Https mode then.

**Comment by ErikMogensen** [comment]
@SeanFeldman, according to https://blogs.msdn.microsoft.com/servicebus/2015/07/10/service-bus-event-hubs-and-web-sockets/ Service Bus (and Event Hubs) uses Web Sockets in Connectivity Mode = Http.

**Comment by SeanFeldman** [comment]
@ErikMogensen indeed. When TCP ports are blocked, the SDK will fail over to HTTPS. 
Will assign this to 4.0.108 and we'll release.

**Comment by SeanFeldman** [comment]
This didn't go out in 4.0.108.
Workaround is to update the ASB client manually and use binding redirect.
Will be addressed in 4.0.109.
